src/rsa/verification.rs

```diff
@@ -282,8 +283,15 @@ pub(crate) fn verify_rsa_(
 
     // Step 3.
     let mut decoded = [0u8; PUBLIC_KEY_PUBLIC_MODULUS_MAX_LEN];
-    let decoded = &mut decoded[..n_bits.as_usize_bytes_rounded_up()];
-    m.fill_be_bytes(decoded);
+    let decoded = {
+        let n_bytes = n_bits.as_usize_bytes_rounded_up();
+        let n_bytes_padded = ((n_bytes + (LIMB_BYTES - 1)) / LIMB_BYTES) * LIMB_BYTES;
+        let decoded = &mut decoded[..n_bytes_padded];
+        m.fill_be_bytes(decoded);
+        let (padding, decoded) = decoded.split_at(n_bytes_padded - n_bytes);
+        assert!(padding.iter().all(|&b| b == 0));
+        decoded
+    };
```

Comment:
I'm not particularly happy with how ugly this is or how leaky the abstractions are. For example, this code is only correct assuming we've done various bounds checks that indicate the RSA modulus length is reasonable.

OTOH, for validation reasons it is important to avoid making the code in `arithmetic` and `limbs` more complicated than it is.

In the future, after the release that contains this fix, we should clean this up.